UrlManager.php

```diff
@@ -356,7 +360,7 @@ protected function redirectToLanguage($language)
 
         Yii::$app->getResponse()->redirect($redirectUrl);
         if (YII_ENV_TEST) {
-            throw new \yii\base\Exception($redirectUrl);
+            throw new \yii\base\Exception(\yii\helpers\Url::to($redirectUrl));
```

Comment:
Oh, now I see. If you call `$response->redirect()` it will not use the raw URL but feed it into `Url::to()` first. Nice catch. This way we make sure, that we really test the correct final URL.